var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -31,16 +31,26 @@ class Ipopt(Package):
     homepage = "https://projects.coin-or.org/Ipopt"
     url      = "http://www.coin-or.org/download/source/Ipopt/Ipopt-3.12.4.tgz"
 
+    version('3.12.7', '2a36e4a04717a8ed7012ac7d1253ae4ffbc1a8fd')
+    version('3.12.6', 'ed4072427fab786fcf6082fe7e6f6c2ed9b5e6f8')
+    version('3.12.5', '3f63ddfff517235ead17af6cceb426ca858dda37')
     version('3.12.4', '12a8ecaff8dd90025ddea6c65b49cb03')
     version('3.12.3', 'c560cbfa9cbf62acf8b485823c255a1b')
     version('3.12.2', 'ec1e855257d7de09e122c446506fb00d')
     version('3.12.1', 'ceaf895ce80c77778f2cab68ba9f17f3')
     version('3.12.0', 'f7dfc3aa106a6711a85214de7595e827')
 
+    variant('coinhsl', default=False,
+            description="Build with Coin Harwell Subroutine Libraries")
+    variant('metis', default=False,
+            description="Build with METIS partitioning support")
+
     depends_on("blas")
     depends_on("lapack")
     depends_on("pkg-config", type='build')
     depends_on("mumps+double~mpi")
+    depends_on('coinhsl', when='+coinhsl')
+    depends_on('metis@4.0.3', when='+metis')
```

Comment:
It only works with METIS 4; see https://www.coin-or.org/Ipopt/documentation/node13.html#SECTION00043700000000000000. I guess, theoretically, the spec could be `depends_on('metis@4')`, and I could try debugging that again, but I think spack returned an error with that particular spec, and `metis@4.0.3` is the only version of METIS 4 in spack.

It's possible to add METIS 4.0.1 to spack, and then test a more general spec, but is doing so really worth it?